doc/api/stream.md

```diff
@@ -443,9 +443,26 @@ first argument. To reliably detect write errors, add a listener for the
 The return value is `true` if the internal buffer does not exceed
 `highWaterMark` configured when the stream was created after admitting `chunk`.
 If `false` is returned, further attempts to write data to the stream should
-stop until the [`'drain'`][] event is emitted. However, the `false` return
-value is only advisory and the writable stream will unconditionally accept and
-buffer `chunk` even if it has not not been allowed to drain.
+stop until the [`'drain'`][] event is emitted.
+
+In case the `false` return value is ignored, the writable stream will
+unconditionally accept and buffer `chunk` even if it has not not been allowed
+to drain. However, this might lead to memory leaks and garbage
```

Comment:
This is what I was trying to say [in the conversation that sparked this PR](https://github.com/nodejs/node/commit/f347dad0b7b1787092cca88789b77eb3def2d319#commitcomment-20351125). If your code leaks when `write()` returns `false`, it is still leaking when `write()` returns `true`. And it is the stream being written to that is doing the leaking. If `write()` returns `false` and never drains itself or you call `end()` after `write()` returns `true` and it never drains its buffer, that’s the same sort of leak (though if the target stream treats `end()` specially, you might only see the leak happen when the target stream fails to drain after `write()` returns `false`).